src/som/primitives/actors/PromisePrims.java

```diff
@@ -184,7 +216,7 @@ protected static final SPromise whenResolvedOrError(final SPromise rcvr,
       assert error.getMethod().getNumberOfArguments() == 2;
 
       Actor current = EventualMessage.getActorCurrentMessageIsExecutionOn();
-      SPromise  promise  = SPromise.createPromise(current);
+      SPromise  promise  = SPromise.createPromise(current, false, false, true);
```

Comment:
maybe that, but more importantly, we want to be able to have a popup there, and set them explicitly.